### PR TITLE
Add dera

### PR DESCRIPTION
**Pull request by marenat**
add DER_A
test to be added

### CODE REVIEW

File: src/models/generated/DER_A.jl [hunk 1]
```diff
@@ -0,0 +1,160 @@
+#=
+This file is auto-generated. Do not edit.
+=#
+
+#! format: off
+
+"""
+    mutable struct DER_A <: InnerControl
```
Comment (jd-lara) [inline src/models/generated/DER_A.jl:8]: The names of struct can't have `_`. We need to pick something more descriptive like `DistributedResourceA` or `ActiveDistributionLoadA`

**Review by jd-lara — CHANGES_REQUESTED** [review]

File: src/models/generated/DER_A.jl (hunk 1 above)
Comment (rodrigomha) [reply src/models/generated/DER_A.jl:8]: My vote goes to ActiveDistributionLoadA or ActiveDistributionNetworkA. I prefer LoadA.

Comment (jd-lara) [reply src/models/generated/DER_A.jl:8]: This is a `<: DynamicInjection`

File: src/descriptors/power_system_structs.json [hunk 2]
```diff
@@ -11565,6 +11565,503 @@
       ],
       "supertype": "InnerControl"
     },
+    {
+      "struct_name": "DER_A",
+      "docstring": "Parameters of the Inner Control part of the REECB model in PSS/E",
```
Comment (jd-lara) [inline src/descriptors/power_system_structs.json:11570]: Incorrect docstring

File: src/descriptors/power_system_structs.json [hunk 3]
```diff
@@ -11565,6 +11565,503 @@
...
+        },
+        {
+          "name": "n_states",
+          "exclude_setter": true,
+          "comment": "The states of the RECurrentControlB depends on the Flags",
+          "internal_default": "2",
+          "data_type": "Int"
+        }
+      ],
+      "supertype": "InnerControl"
```
Comment (jd-lara) [inline src/descriptors/power_system_structs.json:12063]: Change the super type here to `DynamicInjection`

File: src/descriptors/power_system_structs.json [hunk 4]
```diff
@@ -11565,6 +11565,503 @@
...
+          "name": "states",
+          "exclude_setter": true,
+          "comment": "The states of the RECurrentControlB depends on the Flags",
+          "internal_default": "PowerSystems.get_REControlB_states(Q_Flag)",
+          "data_type": "Vector{Symbol}"
+        },
+        {
+          "name": "n_states",
+          "exclude_setter": true,
+          "comment": "The states of the RECurrentControlB depends on the Flags",
```
Comment (jd-lara) [inline src/descriptors/power_system_structs.json:12058]: number of states should be depending on the flag?

File: src/descriptors/power_system_structs.json [hunk 5]
```diff
@@ -11565,6 +11565,503 @@
...
+        {
+          "name": "ext",
+          "data_type": "Dict{String, Any}",
+          "null_value": "Dict{String, Any}()",
+          "default": "Dict{String, Any}()"
+        },
+        {
+          "name": "states",
+          "exclude_setter": true,
+          "comment": "The states of the RECurrentControlB depends on the Flags",
```
Comment (jd-lara) [inline src/descriptors/power_system_structs.json:12051]: change docstring

**Comment by rodrigomha** [comment]
I think a rebase will be needed to fix the conflicts with the includes.jl file

**Review by jd-lara — APPROVED** [review]

**Review by rodrigomha — CHANGES_REQUESTED** [review]
@m-bossart @marenat I think the error is happening because you need to include the file `dynamic_loads.jl` as `include` in the `PowerSystems.jl` file

File: src/models/dynamic_loads.jl [hunk 6]
```diff
@@ -0,0 +1,9 @@
+function get_AggregateDistributedGenerationA_states(Freq_Flag::Int)
+    if Freq_Flag == 0
+        return [:Vmeas, :Pmeas, :Q_V, :Iq, :Mult, :Fmeas, :dPord, :Pord, :Ip], 9
+    elseif Freq_Flag == 1
+        return [:Vmeas, :Pmeas, :Q_V, :Iq, :Mult, :Fmeas, :Power_PI, :dPord, :Pord, :Ip], 10
```
Comment (jd-lara) [inline src/models/dynamic_loads.jl:5]: This line needs testing

File: src/models/dynamic_loads.jl [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+    else
+        error("Unsupported value of Freq_Flag")
```
Comment (jd-lara) [inline src/models/dynamic_loads.jl:7]: Also test this throws an error use `test_throws`

**Review by jd-lara — CHANGES_REQUESTED** [review]
See comments below

**Review by jd-lara — APPROVED** [review]

**Review by rodrigomha — APPROVED** [review]

**Pull request merged**